File: finder4j-backtrack-examples/src/test/java/com/github/K0zka/finder4j/backtrack/examples/queens/QueensStateTest.java
package com.github.K0zka.finder4j.backtrack.examples.queens;

import org.junit.Assert;
import org.junit.Test;

import com.github.K0zka.finder4j.backtrack.examples.queens.QueensState;

public class QueensStateTest {
	@Test
	public void isLegal() {
		QueensState state = new QueensState(new Short[] { 1, null });
		Assert.assertTrue(state.isLegal());
	}

	@Test
	public void isLegalNegative() {
		QueensState state = new QueensState(new Short[] { 1, 1 });
		Assert.assertFalse(state.isLegal());

		state = new QueensState(new Short[] { 1, 2 });
		Assert.assertFalse(state.isLegal());

		state = new QueensState(new Short[] { 1, 3, 1 });
		Assert.assertFalse(state.isLegal());

		state = new QueensState(new Short[] { 1, 5, 3 });
		Assert.assertFalse(state.isLegal());

	}

	@Test
	public void testToString() {
		System.out.println(new QueensState(new Short[] { 1, 5, 7, 2 }));
		System.out.println(new QueensState(new Short[] { 1, 5, 7, null, 1, 5, 7, null }));
	}

}

File: finder4j-backtrack-examples/src/test/kotlin/com/github/k0zka/finder4j/backtrack/examples/queens/QueensStateTest.kt
package com.github.k0zka.finder4j.backtrack.examples.queens

import com.github.k0zka.finder4j.backtrack.examples.queens.QueensState.Companion.notSet
import org.junit.Assert
import org.junit.Test

class QueensStateTest {
	@Test
	fun isLegal() {
		val state = QueensState(arrayOf(1, notSet).toShortArray())
		Assert.assertTrue(state.isLegal)
	}

	@Test
	fun isLegalNegative() {
		var state = QueensState(arrayOf(1.toShort(), 1.toShort()).toShortArray())
		Assert.assertFalse(state.isLegal)

		state = QueensState(arrayOf(1.toShort(), 2.toShort()).toShortArray())
		Assert.assertFalse(state.isLegal)

		state = QueensState(arrayOf(1.toShort(), 3.toShort(), 1.toShort()).toShortArray())
		Assert.assertFalse(state.isLegal)

		state = QueensState(arrayOf(1.toShort(), 5.toShort(), 3.toShort()).toShortArray())
		Assert.assertFalse(state.isLegal)

	}

	@Test
	fun testToString() {
		println(QueensState(arrayOf(1.toShort(), 5.toShort(), 7.toShort(), 2.toShort()).toShortArray()))
		println(QueensState(
				arrayOf(1.toShort(), 5.toShort(), 7.toShort(), notSet, 1.toShort(), 5.toShort(), 7.toShort(),
						notSet).toShortArray()))
	}

}
